prow/cmd/validation-webhook/main.go

```diff
@@ -0,0 +1,94 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/test-infra/experiment/clustersecretbackup/secretmanager"
+	"k8s.io/test-infra/prow/logrusutil"
+)
+
+const (
+	secretID  = "ca-certification"
+	secretName = "ca-secrets"
+	caCert = "ca-cert"
+	caPrivKey = "ca-priv-key"
+	caBundle = "ca-bundle"
+)
+
+func main() {
+	logrusutil.ComponentInit()
+	logrus.SetLevel(logrus.DebugLevel)
+	// GCP project in which to store secrets in Secret Manager
+	p := flag.String("p", 
+	"colew-test", "Project ID for storing GCP Secrets")
+	flag.Parse()
+	projectId := *p
+	secretsMap := make(map[string]string)
+	client, err := secretmanager.NewClient(projectId, false)
+	ctx := context.Background()
+	if err != nil {
+		logrus.WithError(err).Fatal("Unable to create secret manager client")
+	}
+	cert, privKey := getSecrets(client, ctx, secretsMap)
+	_, err = isCertValid(cert)
+	if err != nil {
+		createGCPSecret(client, ctx)
+		cert, privKey = getSecrets(client, ctx, secretsMap)
+	} 
+	runServer(cert, privKey)
+
+}
+
+func getSecrets(client *secretmanager.Client, ctx context.Context, secretsMap map[string]string) (string, string) {
```

Comment:
im unmarshalling the map and returning the cert and privkey if getting the secret succeeds, I could make it global i guess?

Reply:
It could just be a local parameter of the function. We don't actually use the map after `getSecrets` so we don't need a reference to it outside this function.